tests/CMakeLists.txt

```diff
@@ -129,6 +129,7 @@ set(PYBIND11_TEST_FILES
     test_stl.cpp
     test_stl_binders.cpp
     test_tagbased_polymorphic.cpp
+    test_type_caster_bare_interface.cpp
```

Comment:
Final nitpick: `test_type_caster_interface` also works, if we want to have this file show off the interface?